lib/uploadex.ex

```diff
@@ -26,11 +26,11 @@ defmodule Uploadex do
 
   This function also deletes files that are no longer referenced.
   """
-  def update_with_file(changeset, previous_record, uploader) do
+  def update_with_file_delete_previous(changeset, previous_record, uploader, opts \\ []) do
```

Comment:
Call this function `update_with_file` and the other one `update_with_file_keep_previous`, because that other function is the one I expect to be called less often